streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -620,22 +620,35 @@ public final TopologyBuilder copartitionSources(Collection<String> sourceNodes)
      * Returns the copartition groups.
      * A copartition group is a group of source topics that are required to be copartitioned.
      *
+     * @param applicationId
      * @return groups of topic names
      */
-    public Collection<Set<String>> copartitionGroups() {
+    public Collection<Set<String>> copartitionGroups(final String applicationId) {
```

Comment:
@guozhangwang if we add this to the `TopologyBuilder` constructor we should probably remove it as a config in `StreamsConfig` otherwise we will be duplicating the applicationId and potentially introducing errors. i.e., currently we do:

```
TopologyBuilder topologyBuilder = new TopologyBuilder();
Properties streamsConfiguration = new Properties();
streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG,
                                 applicationId);
streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG,CLUSTER.bootstrapServers());
...
new KafkaStreams(topologyBuilder, streamsConfiguration)
```

With this change:

```
TopologyBuilder topologyBuilder = new TopologyBuilder(applicationId);
streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG,CLUSTER.bootstrapServers());
...
new KafkaStreams(topologyBuilder, streamsConfiguration)
```